ui/app/rif/components/chainAddresses.js

```diff
@@ -130,14 +130,17 @@ class ChainAddresses extends Component {
     this.props.waitForListener(transactionListenerId)
       .then(async (transactionReceipt) => {
         if (this.state.resolvers.find(resolver => resolver.address === this.props.selectedResolverAddress)) {
-          await this.props.deletePendingChainAddress(selectedChainAddress, !!this.props.subdomainName);
-          const chainAddresses = await this.props.getChainAddresses(this.props.domainName, this.props.subdomainName);
-          this.props.showThis(
-            this.props.redirectPage,
-            {
-              ...this.props.redirectParams,
-              newChainAddresses: chainAddresses,
-            });
+          // This timeout is here because as we are using the notifier service, when we recieve the success, the notifier still doesnt have the last notification
```

Comment:
Can you explain a little more this? 

What about if the notification isnt present after 5 secs?

Reply:
Now this was moved to a function, and it's auto-calling itself if the array given has no difference